includes/Checker/Abstract_Check_Runner.php

```diff
@@ -186,14 +205,40 @@ final public function set_experimental_flag( $include_experimental ) {
 		if ( $this->initialized_early ) {
 			if ( $include_experimental !== $this->get_include_experimental_param() ) {
 				throw new Exception(
-					__( 'Invalid flag: The include-experimental flag does not match the original request parameter.', 'plugin-check' )
+					sprintf(
+						/* translators: %s: include-experimental */
+						__( 'Invalid flag: The %s flag does not match the original request parameter.', 'plugin-check' ),
+						'include-experimental'
+					)
 				);
 			}
 		}
 
 		$this->include_experimental = $include_experimental;
 	}
 
+	/**
+	 * Sets categories for filtering the checks.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $categories An array of categories for filtering.
+	 *
+	 * @throws Exception Thrown if the getegories does not match the original request parameter.
+	 */
+	final public function filter_checks_by_specific_categories( $categories ) {
```

Comment:
The method name here is not right. This sets the check categories, it doesn't filter anything. It should be called `set_categories()`.